tests/msc3030_test.go

```diff
@@ -203,6 +203,16 @@ func TestJumpToDateEndpoint(t *testing.T) {
 				contextResResBody := client.ParseJSON(t, contextRes)
 				paginationToken := client.GetJSONFieldStr(t, contextResResBody, "end")
 
+				// Hit `/messages` until `eventA` has been backfilled and replicated across
+				// workers.
+				fetchUntilMessagesResponseHas(t, remoteCharlie, roomID, func(ev gjson.Result) bool {
```

Comment:
A comment explaining why eventA doesn't appear in the first /messages response would be useful

Reply:
I feel like it already has it: `until 'eventA' has been backfilled and replicated across workers.`

What aspects would make this more clear into the future? I assume you're after trying to boil down the "Why did this test fail with workers before?" section from the description. Specific language suggestion welcome! Maybe something around not having ["read-after-write consistency"](https://github.com/matrix-org/synapse/issues/13185#issuecomment-1177930853) which @DMRobertson mentioned.

Reply:
We could say something like "the worker doing the backfilling isn't necessarily the same as the workerserving /messages"? :shrug: Tricky really: complement is supposed to be HS agnostic and not a description of Synapse's quirks.

I wouldn't worry too much about it: let's not block this too much on wording.

Reply:
Added 👍 - Good wording @DMRobertson!